### PR TITLE
Correct auto-generated header line.

### PR DESCRIPTION
**Pull request by adam-azarchs**
The header line noting that the file is automatically generated
should conform to [established conventions](https://golang.org/cmd/go/#hdr-Generate_Go_files_by_processing_source) for indicating that
a file is automatically generated.

Ideally it should also not hard-coded the header to 10X Genomics,
since this is an open-source tool that is not developed from within
the 10X Genomics github organization.

There will probably need to be other changes made to finish this PR.

### CODE REVIEW

**Comment by sreenathkrishnan** [comment]
Cool! Perhaps the copyright header can be specified by the code that calls the mro generator function?

**Comment by adam-azarchs** [comment]
Yes, that would be best I think.

File: martian/src/lib.rs [hunk 1]
```diff
@@ -300,8 +305,31 @@ pub fn make_mro_string(mro_registry: &[StageMro]) -> String {
...
+
+{}{}{}",
+            get_generator_name(),
+            filetype_header,
+            struct_header,
+            mro_string
+        )
+    } else {
+        format!(
+            "{}
```
Comment (sreenathkrishnan) [inline martian/src/lib.rs:322]: It might be good to check that all the lines in `header_comment` start with `#` here?

**Review by sreenathkrishnan — APPROVED** [review]
Looks great except for the one comment about checking the validity of the header comment 👍

File: martian/src/lib.rs (hunk 1 above)
Comment (adam-azarchs) [reply martian/src/lib.rs:322]: Honestly not terribly interested in figuring out how to do that in rust; as far as I'm concerned if a user wants to generate invalid MRO they're welcome to do so.  Open to suggestions, though.

Comment (adam-azarchs) [reply martian/src/lib.rs:330]: Done.  Please check if I did it properly.

Comment (sreenathkrishnan) [reply martian/src/lib.rs:330]: Awesome! Looks good to me.

**Pull request merged**